libnmstate/nm/wired.py

```diff
@@ -92,24 +93,24 @@ def create_setting(iface_state, base_con_profile):
             nm_wired_setting.props.duplex = None
 
         elif not setting.speed:
-            ethtool_results = minimal_ethtool(str(iface_state['name']))
-            setting.speed = ethtool_results['speed']
+            ethtool_results = minimal_ethtool(str(iface_state[Interface.NAME]))
+            setting.speed = ethtool_results[Ethernet.SPEED]
         elif not setting.duplex:
-            ethtool_results = minimal_ethtool(str(iface_state['name']))
-            setting.duplex = ethtool_results['duplex']
+            ethtool_results = minimal_ethtool(str(iface_state[Interface.NAME]))
+            setting.duplex = ethtool_results[Ethernet.DUPLEX]
 
     elif setting.auto_negotiation is False:
         nm_wired_setting.props.auto_negotiate = False
-        ethtool_results = minimal_ethtool(str(iface_state['name']))
+        ethtool_results = minimal_ethtool(str(iface_state[Interface.NAME]))
         if not setting.speed:
-            setting.speed = ethtool_results['speed']
+            setting.speed = ethtool_results[Ethernet.SPEED]
         if not setting.duplex:
-            setting.duplex = ethtool_results['duplex']
+            setting.duplex = ethtool_results[Ethernet.DUPLEX]
 
     if setting.speed:
         nm_wired_setting.props.speed = setting.speed
 
-    if setting.duplex in ['half', 'full']:
+    if setting.duplex in [Ethernet.HALF_DUPLEX, Ethernet.FULL_DUPLEX]:
```

Comment:
sorry, another nitpick: It is probably better to make full duplex the first setting since I believe it will be used more often. Not sure it is worth the effort but it might be a possible optimization (with little effect probably).

Reply:
Sorry, too late. Please send a patch, it sounds fine to me.